algorithms/strings.py

```diff
@@ -0,0 +1,33 @@
+# https://leetcode.com/problems/buddy-strings/
+class Solution:
+    def buddyStrings(self, A: str, B: str):
+        if len(A) != len(B):
+            return False
+        swap = []
+        sim_chars = {}
+        s = 0
+        i = 0
+
+        while i < len(A):
+            if A[i] not in sim_chars:
+                sim_chars[A[i]] = 0
```

Comment:
if I understand correctly, you don't need a dict, really, but just a set - you never increment a value, just using a `not in` check.
This probably won't affect the performance, but might make code a bit clearer.